### DISCUSSION TITLE
Hide the titlebar icon

### DISCUSSION THREAD

**Discussion by Stono** [ghostty-org/ghostty#3343] [Q&A] [answered]
Hi, 
This is a super minor petty thing but i hate emojis :D I'm trying to find a way to hide the ghost emoji in the title bar

<img width="1056" alt="Screenshot 2024-12-27 at 08 42 25" src="https://github.com/user-attachments/assets/adf1e401-5765-41b2-a1cd-8148fe0a9705" />

I've had a good look through the docs and can't see a way to do this without hiding the entire title bar, which I don't want to do.

Am I being silly? Is there a way to do this?

**Comment by 00-kat** [comment]
If I remember correctly, the ghost emoji is just set with `title`:
```ini
# The title Ghostty will use for the window. This will force the title of the
# window to be this title at all times and Ghostty will ignore any set title
# escape sequences programs (such as Neovim) may send.
# 
# If you want a blank title, set this to one or more spaces by quoting
# the value. For example, `title = " "`. This effectively hides the title.
# This is necessary because setting a blank value resets the title to the
# default value of the running program.
# 
# This configuration can be reloaded at runtime. If it is set, the title
# will update for all windows. If it is unset, the next title change escape
# sequence will be honored but previous changes will not retroactively
# be set. This latter case may require you restart programs such as neovim
# to get the new title.
title =
```

**Comment by Stono** [comment]
Gave that a go, setting it to " " removed the title text but leaves me with a folder icon instead lol:

<img width="834" alt="Screenshot 2024-12-27 at 11 55 45" src="https://github.com/user-attachments/assets/7c7f6881-72e4-4b9b-b0cf-abe3dd308b90" />

Setting it to `title =` seems to give me the last command run (plus an icon):

<img width="821" alt="Screenshot 2024-12-27 at 11 56 35" src="https://github.com/user-attachments/assets/4e7ce31e-f408-4cce-97d9-f8ba25fdf15b" />

**Reply by gpanders** [reply]
The folder icon is the macOS proxy icon. If you don’t want it you can disable it with https://ghostty.org/docs/config/reference#macos-titlebar-proxy-icon

**Comment by Stono** [comment]
Ahhh its shell integration, so doing: 

```
shell-integration = none
title = " "
```

_fixes_ it, now i just need to understand what else i lose by disabling shell integration!

**Comment by SuperDuperDeou** [comment]
You can set `shell-integration-features = cursor,sudo,no-title` or just `shell-integration-features = no-title` to keep all the shell integration features but disable the title icon.

**Comment by Stono** [comment]
`macos-titlebar-proxy-icon` ! jackpot! Thank you @gpanders  :D

**Comment by huhu415** [comment, marked as answer]
```
macos-titlebar-proxy-icon = hidden
title = " "
```

The effect is as follows
<img width="912" alt="截屏2025-01-03 17 27 30" src="https://github.com/user-attachments/assets/bb13ca9b-a1b3-43d6-b6bd-99dc41d6f9db" />

**Reply by rnmp** [reply]
this is the correct answer

**Reply by jumbosushi** [reply]
thanks! I had to restart ghostty for the icon to disappear

**Reply by Proteusiq** [reply]
Can you share you config? I have restarted ghostty and mine persisted

**Reply by rauf322** [reply]
how did you make your title bar the same color as the whole terminal?

**Reply by 00-kat** [reply]
[`macos-titlebar-style = tabs`] I think.

[`macos-titlebar-style = tabs`]: https://ghostty.org/docs/config/reference#macos-titlebar-style

**Comment by Proteusiq** [comment]
![CleanShot 2025-01-11 at 15 04 10@2x](https://github.com/user-attachments/assets/482c0c30-8665-4b97-8858-1fba89501091)
None of that works for me

```sh
macos-titlebar-proxy-icon = hidden
title = " "
```

![CleanShot 2025-01-11 at 15 06 41@2x](https://github.com/user-attachments/assets/0a35bb72-2abd-45f8-a5c4-5584b2b8ce50)

**Reply by 00-kat** [reply]
```ini
shell-integration-features = no-title
```
?

**Reply by Proteusiq** [reply]
`shell-integration-features = no-title`

![image](https://github.com/user-attachments/assets/0c4b2ba6-d23d-4a1e-a593-e359e5600f00)

Nope

**Comment by unmovingcastle** [comment]
I have the exact opposite problem, I really like the ghost emoji.
How do I change the proxy-icon to **always** be the ghost?

**Reply by 00-kat** [reply]
If you don't have the same problem, you should open a new Discussion.

Anyways, you either want to force it to always be the ghost:
```ini
title = 👻
```
Or disable the part of shell integration which overrides it:
```ini
shell-integration-features = no-title
```

The former will disallow programs running in the terminal emulator from overriding it, while the latter would only disable the thing that makes it not show a ghost.
